provider/cluster/kube/client.go

```diff
@@ -710,13 +747,97 @@ func (rp resourcePair) available() resource.Quantity {
 }
 
 type nodeResources struct {
-	cpu     resourcePair
-	memory  resourcePair
-	storage resourcePair
-	arch    string
+	cpu                       resourcePair
+	memory                    resourcePair
+	ephemeralStorage          resourcePair
+	persistentVolumes         resourcePair
+	arch                      string
+	volumesAttached           resourcePair
+	persistentVolumesAttached resourcePair
+	volumesMounted            resourcePair
+	persistentVolumesMounted  resourcePair
+	storageclasses            map[string]bool
+}
+
+type NodeStorage struct {
+}
+
+type ActiveClusterStorage struct {
+}
+
+func (c *client) countKubeCall(err error, name string) {
```

Comment:
does this need to be part of the client ?

Reply:
it's being sitting there for a while. it should be middleware to k8s rest client